nomad/helper/import.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package helper
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var (
+	missingNamespaceImportErr = errors.New("missing namespace, the import ID should follow the pattern <id>@<namespace>")
+	missingIDImportErr        = errors.New("missing resource ID, the import ID should follow the pattern <id>@<namespace>")
+)
+
+// NamespacedImporterContext imports a namespaced resource that doesn't have
+// its namespace as part of the Terraform resource ID.
+func NamespacedImporterContext(_ context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	namespacedID := d.Id()
+	sepIdx := strings.LastIndex(d.Id(), "@")
```

Comment:
super nit: you defined `namespacedID` just now but then didn't use it :P

```suggestion
	sepIdx := strings.LastIndex(namespacedID, "@")
```